### ISSUE TITLE
Renaming Any Bone in Armature Causes Script to Fail

### ISSUE THREAD

**Issue by PandaDara** [lyuma/Av3Emulator#5]
So, I had an avatar that had a strange issue where the Chest bone would get renamed to Chest 1 in Unity despite it being named correctly in Blender. I figured I would do a minor change and just rename the bone to Chest in Unity and everything seemed fine, things moved properly. Later on, I noticed that the AV3 emulator would fail, on runtime when I entered play mode the component added to the avatar to control it would be disabled and no parameters would be present to change. Enabling it and changing values had no apparent effect on the model.

tldr; don't change any bone names in Unity.

**Comment by lyuma** [comment]
Very strange! I have no idea why the emulator would fail to start aside from if your newly rigged model wasn't Humanoid. One thing I'll suggest: can you make sure the model is properly rigged as Humanoid and not Generic?

I fixed an error preventing Generic models from working. But my guess is you wanted Humanoid anyway.

Also, if you keep getting the error, would it be possible to check your Console window for any red errors. If there's a lot, stop playing and scroll up to the first few red errors?

Feel free to DM me Lyuma #0781 if you want to debug this live and are still stuck.
